### PR TITLE
[Windows, Keyboard] Enqueue keyboard message 

### PR DESCRIPTION
**Pull request by dkwingsmt**
This PR changes how Win32 processes keyboard messages. The messages are now put into a queue, and the next message will not be processed at all until the previous message has been responded by the framework (or if the message does not need a response, such as a single `WM_CHAR`).

This fixes a racing condition described in https://github.com/flutter/flutter/issues/82673#issuecomment-1072799572. A similar strategy has been applied to macOS in https://github.com/flutter/engine/pull/32120.

A previous attempt, https://github.com/flutter/engine/pull/31379, alleviates the problem but failed to handle more complex scenarios.

A few unit tests have to be altered. Previously later key events will be dispatched before the framework responds to the first event. Now they will only be dispatched after the response. No perceptive changes are made.

A unit test "DisorderlyRespondedEvents" is removed, because it will no longer happen. Events will always be responded in order.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/keyboard_manager_win32.cc [hunk 1]
```diff
@@ -338,38 +252,64 @@ bool KeyboardManagerWin32::HandleMessage(UINT const action,
...
-      //
-      // Messages with Control or Win modifiers down are never considered as
-      // character messages. This allows key combinations such as "CTRL + Digit"
-      // to properly produce key down events even though `MapVirtualKey` returns
-      // a valid character. See https://github.com/flutter/flutter/issues/85587.
+      // Check if this key produces a character by peeking if this key down
+      // message has a following char message. Certain key messages are not
+      // followed by char messages even though `MapVirtualKey` returns a valid
+      // character (such as Ctrl + Digit, see
+      // https://github.com/flutter/flutter/issues/85587 .)
```
Comment (gspencergoog) [inline shell/platform/windows/keyboard_manager_win32.cc:296]: Super-ultra nit: :-)

```suggestion
      // https://github.com/flutter/flutter/issues/85587 ).
```

File: shell/platform/windows/keyboard_win32_unittests.cc [hunk 2]
```diff
@@ -1954,57 +1954,6 @@ TEST(KeyboardTest, UpOnlyImeEventsAreCorrectlyHandled) {
   clear_key_calls();
 }
 
-TEST(KeyboardTest, DisorderlyRespondedEvents) {
```
Comment (gspencergoog) [inline shell/platform/windows/keyboard_win32_unittests.cc:1957]: Wouldn't you still want to test that events that used to arrive before the framework response no longer do?  Or are the other test changes sufficient to guarantee that?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: shell/platform/windows/keyboard_win32_unittests.cc (hunk 2 above)
Comment (dkwingsmt) [reply shell/platform/windows/keyboard_win32_unittests.cc:1957]: Other test changes are sufficient to guarantee that even when the platform messages arrive too fast for the framework to handle, nothing will be responded out of order and the events turn out to be normal.

**Pull request merged**